api/server/router/container/container_routes.go

```diff
@@ -609,17 +608,30 @@ func (s *containerRouter) postContainersCreate(ctx context.Context, w http.Respo
 		}
 	}
 
-	if config != nil && versions.LessThan(version, "1.44") {
+	var deprecatedMacAddress string
+	if config != nil {
+		deprecatedMacAddress = config.MacAddress
+	}
+
+	var warnings []string
+	if versions.LessThan(version, "1.44") {
 		// The container-wide MacAddress parameter is deprecated and should now be specified in EndpointsConfig.
-		if config.MacAddress != "" && (hostConfig.NetworkMode.IsDefault() || hostConfig.NetworkMode.IsBridge() || //nolint:staticcheck
-			hostConfig.NetworkMode.IsUserDefined()) {
+		if hostConfig != nil && (hostConfig.NetworkMode.IsDefault() || hostConfig.NetworkMode.IsBridge() || hostConfig.NetworkMode.IsUserDefined()) {
 			nwName := hostConfig.NetworkMode.NetworkName()
-			if _, ok := networkingConfig.EndpointsConfig[nwName]; !ok {
-				networkingConfig.EndpointsConfig[nwName] = &network.EndpointSettings{}
+			if endPointConfig, ok := networkingConfig.EndpointsConfig[nwName]; !ok {
```

Comment:
Few things I'm gonna fix:

1. `networkingConfig` could be nil ;
2. We should invariably set the endpoint's `MacAddress`, as it'll be either the container-wide MacAddress, or an empty string.